### ISSUE TITLE
.awikstats misc issues

### ISSUE THREAD

**Issue by jonorthwash** [apertium/phenny#486]
I ran `.awikstats update kaz-kaa` and [this is what it did](http://wiki.apertium.org/w/index.php?title=Apertium-kaz-kaa%2Fstats&diff=69841&oldid=51501).

Issues I see:
- The username `francis.m..tyers` is not a github username, nor is it even a valid part of one.  It should be @ftyers, I think.
- It added the incorrect `rlx rlx rules` instead of updating `rlx rules`.
- It added `dix stems` instead of updating (/renaming) `stems`.
- Links are to raw files at a given commit (e.g., https://raw.githubusercontent.com/apertium/apertium-kaz-kaa/034cb6038dac0e0d9428f25fe1579c954276eb38/apertium-kaz-kaa.kaz-kaa.rlx) instead of a github page with the diff of the file at that commit (e.g., https://github.com/apertium/apertium-kaz-kaa/commit/034cb6038dac0e0d9428f25fe1579c954276eb38#diff-593d5b5a1eb858097a6f44040ef3cb03).

**Comment by sushain97** [comment]
> * The username francis.m..tyers is not a github username, nor is it even a valid part of one. It should be @ftyers, I think.

This is as designed but a bit unfortunate. The username's are GitHub's translation of a GH user to an SVN user. https://github.com/apertium/apertium-stats-service/issues/52 should be able to modify the behavior.

> * It added the incorrect `rlx rlx rules` instead of updating `rlx rules`.
> * It added `dix stems` instead of updating (/renaming) `stems`.

This sounds like bugs in this repo.

> * Links are to raw files at a given commit (e.g., https://raw.githubusercontent.com/apertium/apertium-kaz-kaa/034cb6038dac0e0d9428f25fe1579c954276eb38/apertium-kaz-kaa.kaz-kaa.rlx) instead of a github page with the diff of the file at that commit (e.g., [apertium/apertium-kaz-kaa@034cb60#diff-593d5b5a1eb858097a6f44040ef3cb03](https://github.com/apertium/apertium-kaz-kaa/commit/034cb6038dac0e0d9428f25fe1579c954276eb38#diff-593d5b5a1eb858097a6f44040ef3cb03)).

This one is definitely by design. Not sure how to generate the latter URL since I'm not sure what the `593d5b5a1eb858097a6f44040ef3cb03` is referring to....

**Comment by jonorthwash** [comment]
> > * The username francis.m..tyers is not a github username, nor is it even a valid part of one. It should be @ftyers, I think.
> 
> This is as designed but a bit unfortunate. The username's are GitHub's translation of a GH user to an SVN user. [apertium/apertium-stats-service#52](https://github.com/apertium/apertium-stats-service/issues/52) should be able to modify the behavior.

With the two dots also?

> 
> > * It added the incorrect `rlx rlx rules` instead of updating `rlx rules`.
> > * It added `dix stems` instead of updating (/renaming) `stems`.
> 
> This sounds like bugs in this repo.

By "this repo" do you mean `phenny` or `apertium-kaz-kaa`?  If the latter, what's the relationship between what's in the repo and what it prints for rule counts?

> 
> > * Links are to raw files at a given commit (e.g., https://raw.githubusercontent.com/apertium/apertium-kaz-kaa/034cb6038dac0e0d9428f25fe1579c954276eb38/apertium-kaz-kaa.kaz-kaa.rlx) instead of a github page with the diff of the file at that commit (e.g., [apertium/apertium-kaz-kaa@034cb60#diff-593d5b5a1eb858097a6f44040ef3cb03](https://github.com/apertium/apertium-kaz-kaa/commit/034cb6038dac0e0d9428f25fe1579c954276eb38#diff-593d5b5a1eb858097a6f44040ef3cb03)).
> 
> This one is definitely by design. Not sure how to generate the latter URL since I'm not sure what the `593d5b5a1eb858097a6f44040ef3cb03` is referring to....

If you click on it, it anchors to the right file.  It's also what's linked to for the commit under [the file's history](https://github.com/apertium/apertium-kaz-kaa/commits/master/apertium-kaz-kaa.kaz-kaa.rlx).  I think even loading that page of diffs for the commit and not having the anchor to the specific file would be preferable to what we have now, since it's in the context of clicking on the commit hash, not the filename.

**Comment by sushain97** [comment]
> > > * The username francis.m..tyers is not a github username, nor is it even a valid part of one. It should be @ftyers, I think.
> > 
> > 
> > This is as designed but a bit unfortunate. The username's are GitHub's translation of a GH user to an SVN user. [apertium/apertium-stats-service#52](https://github.com/apertium/apertium-stats-service/issues/52) should be able to modify the behavior.
> 
> With the two dots also?

Yep.



> > > * It added the incorrect `rlx rlx rules` instead of updating `rlx rules`.
> > > * It added `dix stems` instead of updating (/renaming) `stems`.
> > 
> > 
> > This sounds like bugs in this repo.
> 
> By "this repo" do you mean `phenny` or `apertium-kaz-kaa`? If the latter, what's the relationship between what's in the repo and what it prints for rule counts?

`phenny`. This repo doesn't do any of the calculation, it just gets the numbers from `stats-service` and updates the wiki. Numbers: https://apertium.projectjj.com/stats-service/apertium-kaz-kaa.

> > > * Links are to raw files at a given commit (e.g., https://raw.githubusercontent.com/apertium/apertium-kaz-kaa/034cb6038dac0e0d9428f25fe1579c954276eb38/apertium-kaz-kaa.kaz-kaa.rlx) instead of a github page with the diff of the file at that commit (e.g., [apertium/apertium-kaz-kaa@034cb60#diff-593d5b5a1eb858097a6f44040ef3cb03](https://github.com/apertium/apertium-kaz-kaa/commit/034cb6038dac0e0d9428f25fe1579c954276eb38#diff-593d5b5a1eb858097a6f44040ef3cb03)).
> > 
> > 
> > This one is definitely by design. Not sure how to generate the latter URL since I'm not sure what the `593d5b5a1eb858097a6f44040ef3cb03` is referring to....
> 
> If you click on it, it anchors to the right file. It's also what's linked to for the commit under [the file's history](https://github.com/apertium/apertium-kaz-kaa/commits/master/apertium-kaz-kaa.kaz-kaa.rlx). I think even loading that page of diffs for the commit and not having the anchor to the specific file would be preferable to what we have now, since it's in the context of clicking on the commit hash, not the filename.

Hmm, yeah, I see what the point of the anchor is. Unfortunately, I don't actually know what that value is and how to generate it programatically. As for going to the diff instead of the file, I think we were replicating the old behavior. However, the diff is doable as well. That would be a change in `phenny` (not a large one I think).

**Comment by jonorthwash** [comment]
> This is as designed but a bit unfortunate.  The username's are GitHub's translation of a GH user to an SVN user. apertium/apertium-stats-service#52 should be able to modify the behavior.

At http://wiki.apertium.org/wiki/Apertium-kaz-kir/stats I'm getting myself as both "jonorthwash" and "jonathan.north.washington".  Are you sure this is as designed?

**Comment by jonorthwash** [comment]
Some action items so far:

- [ ] Have phenny add `rlx rules` instead of `rlx rlx rules`.
- [ ] Have phenny update (or rename) `stems` instead of adding `dix stems`.
- [ ] Have links for commit hashes point to the page of diffs for that commit instead of to the raw file at that commit.

**Comment by jonorthwash** [comment]
Also, it looks like transfer macros aren't being counted correctly.  The [kaz-kir stats page](http://wiki.apertium.org/wiki/Apertium-kaz-kir/stats) reports 0 kaz-kir t1x macros, but [kaz-kir.t1x](https://github.com/apertium/apertium-kaz-kir/blob/master/apertium-kaz-kir.kaz-kir.t1x) has 6.

**Comment by sushain97** [comment]
> At http://wiki.apertium.org/wiki/Apertium-kaz-kir/stats I'm getting myself as both "jonorthwash" and "jonathan.north.washington". Are you sure this is as designed?

Yep. This seems attributable to different emails being used which appears is what GH uses to create the SVN authors (at least partially)? It's a bit unclear. Note the two emails below for you.

```
$ git log --pretty=format:"%ae" | sort | uniq | cut -d '@' -f 1
ftyers
ilnar
itang1
jonathan.n.washington
jonathan.north.washington
kakubay
kantoro90
sushain
unhammer
```

> Also, it looks like transfer macros aren't being counted correctly. The [kaz-kir stats page](http://wiki.apertium.org/wiki/Apertium-kaz-kir/stats) reports 0 kaz-kir t1x macros, but [kaz-kir.t1x](https://github.com/apertium/apertium-kaz-kir/blob/master/apertium-kaz-kir.kaz-kir.t1x) has 6.

Hmm, this one sounds like a bug. Does this look like the right way to count them?

https://github.com/apertium/apertium-stats-service/blob/master/src/stats/xml.rs#L95

IIRC, I copied whatever the old Python scripts did.

**Comment by jonorthwash** [comment]
> Does this look like the right way to count them?

No, it should be counting `<def-macro ...>...</def-macro>` blocks.

**Comment by sushain97** [comment]
Ah... I wonder why I wrote it the other way. Fixed: https://github.com/apertium/apertium-stats-service/commit/ac4cbacb34f0eed65c6920b8e84897f8229794ae.

I'll deploy it in a bit.

**Comment by sushain97** [comment]
@jonorthwash look any better?

**Comment by jonorthwash** [comment]
The raw data at https://apertium.projectjj.com/stats-service/apertium-kaz-kir looks good now; I'll have to check the output of `.awikstats` later.
